src/dependency_injector/wiring.py

```diff
@@ -598,14 +598,14 @@ def _locate_dependent_closing_args(provider: providers.Provider) -> Dict[str, pr
         return {}
 
     closing_deps = {}
-    for arg in provider.args:
+    for arg in [*provider.args, *provider.kwargs.values()]:
         if not isinstance(arg, providers.Provider) or not hasattr(arg, "args"):
             continue
-
-        if not arg.args and isinstance(arg, providers.Resource):
+        if isinstance(arg, providers.Resource):
             return {str(id(arg)): arg}
```

Comment:
Hello. Fix not working for several Resources. Maybe something like this?
```suggestion
            closing_deps |= {str(id(arg)): arg}
```